nav2_bringup/launch/localization_launch.py

```diff
@@ -146,11 +162,21 @@ def generate_launch_description():
                 target_container=container_name,
                 composable_node_descriptions=[
                     ComposableNode(
+                        condition=LaunchConfigurationEquals('map', ''),
```

Comment:
Not sure, perhaps at the `LoadComposableNodes` level instead of the `ComposableNode` level? So there would be 3 separate loads: 1 with map reconfigured, 1 without, and 1 for everything else

Reply:
It looks like the composable node function is a ["description"](https://github.com/ros2/launch_ros/blob/7b0aa892662cf1e2d2a6a5ffc4fa67ed9f834410/launch_ros/launch_ros/descriptions/composable_node.py#L36) whereas the node function is an ["action"](https://github.com/ros2/launch_ros/blob/7b0aa892662cf1e2d2a6a5ffc4fa67ed9f834410/launch_ros/launch_ros/actions/node.py#L113). Load Composable nodes is an ["action"](https://github.com/ros2/launch_ros/blob/7b0aa892662cf1e2d2a6a5ffc4fa67ed9f834410/launch_ros/launch_ros/actions/load_composable_nodes.py#L55) so maybe that has the `condition` field

Reply:
The [descriptor](https://github.com/ros2/launch_ros/blob/7b0aa892662cf1e2d2a6a5ffc4fa67ed9f834410/launch_ros/launch_ros/descriptions/composable_node.py#L60,) also has a condition param, I am guessing it should to be possible to use our condition but in its specific format of type [SomeSubstitutionsType ](https://github.com/ros2/launch/blob/rolling/launch/launch/some_substitutions_type.py). I will look into it, if not try the load composable nodes solution.

Reply:
Related issue, bug in composablenode launch: https://github.com/ros2/launch_ros/issues/114

Reply:
> Not sure, perhaps at the `LoadComposableNodes` level instead of the `ComposableNode` level? So there would be 3 separate loads: 1 with map reconfigured, 1 without, and 1 for everything else

The current solution uses this proposal and it works. The issue with the condition param of the composablenode not working should have been fixed in https://github.com/ros2/launch_ros/issues/114 but for some reason it does not work. I am not sure if it has something to do with the ros2 version.

Reply:
Or it was broken again by something, mhm.